src/target/riscv/riscv.c

```diff
@@ -852,24 +859,26 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 		struct trigger_request_info lt_1 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.enable |
 				fields.match.lt,
-			.tdata2 = trigger->address,
+			.tdata2 = trigger->address + trigger->length,
```

Comment:
Could you add a test case for this code in riscv-tests/debug? I believe spike supports these trigger types already.

Reply:
On spike at least, all triggers are the same. Every trigger supports all the features.

Agreed that the test is separate, but it would have caught the bug you're fixing so it seems good to have coverage, given how simple the test (presumably) will be.